### ISSUE TITLE
Consistent Flatpak pinning support in ITM when debian version of application is present.

### ISSUE THREAD

**Issue by jaszhix** [linuxmint/Cinnamon#7974]
Moved from jaszhix/icingtaskmanager#229, @smurphos wrote: 
#
 - Cinnamon 3.8.9
 - Extension version 6.3.13
 - Linux distribution Mint 19

Hi jaszhiz,

There a are a few inconsistencies in the way ITM handles flatpak app pinning when there is also a debian packaged version of the app installed which I hope can be resolved and also included in the fork ADW for Cinnamon 4. This issue is also partly prompted by this 18.3 users experience with snap package pinning - https://github.com/jaszhix/icingtaskmanager/issues/228, which was also raised on the Mint forum. I'm not sure if the same underlying issues cause the inconsistency with snap.

For this testing I've used VLC

Preinstalled version is 3.03 - desktop file is usr/share/applications/vlc.desktop

Flatpak version is 3.04 - desktop file is /var/lib/flatpak/app/org.videolan.VLC/current/active/export/share/applications/org.videolan.VLC.desktop
There is also a symlink in /var/lib/flatpak/exports/share/applications/org.videolan.VLC.desktop

Test 1 - Launch VLC flatpack from the menu, right click on icon and select pin to panel.
Result - vlc.desktop is pinned
Preferred result - org.videolan.VLC.desktop should be pinned.

Test 2 - Right click Flatpak VLC in menu and select add to panel.
Result - org.videolan.VLC.desktop is pinned
Launching from the pinned icon launches the correct version of VLC but another VLC icon is added to ITM, it doesn't group/count the window under the pinned org.videolan.VLC.desktop
Preferred result - Correct grouping of the window.

Test 3 - Use the menu editor to rename the Flatpack VLC entry to VLC Flatpak. This creates a new desktop file - _/.local/share/applications/org.videolan.VLC.desktop
Repeat test 1 - same result
Repeat test 2 - same result

Test 4 - Use the menu-editor to create a new user added menu entry targeting the flatpak version.
Repeat test 1 - same result
Repeat test 2 - same result

Test 5 - Use a different flatpak without a corresponding debian installed version. I used Oomox for testing - desktop file is /var/lib/flatpak/app/com.github.themix_project.Oomox/current/active/export/share/applications/com.github.themix_project.Oomox.desktop with a symlink in /var/lib/flatpak/exports/share/applications/com.github.themix_project.Oomox.desktop

Retried test 1 - works fine with no unexpected results.
Retried test 2 - works fine with no unexpected results.

Contents of org.videolan.VLC.desktop https://pastebin.com/wme6TLp1
Contents of vlc.desktop https://pastebin.com/1ey72Hv4

**Comment by jaszhix** [comment]
Similar behavior is also occurring with the menu. From the original issue

> I'm finding a bug in the menu where I can't even launch the Flatpak version of the app if the Debian version is running and vice versa. More people can analyze it there, and a proper solution probably won't exist in applet code.

In this case GIMP 2.8 and GIMP 2.10, installed via mintinstall.

**Comment by smurphos** [comment]
Yep, menu launching bug reproduced for me with VLC. If i add both launchers to the desktop the bug also occurs trying to run from there.

Partial workaround - amend the command line for the flatpack launcher from....

`/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=/app/bin/vlc --file-forwarding org.videolan.VLC --started-from-file @@u %U @@`

to

`/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=/app/bin/vlc --file-forwarding org.videolan.VLC`

And the flatpak can be launched when the debian version is running, but not vice-versa.

This isn't Cinnamon specific - then menu / desktop launcher bug is reproducible in Mate 19.

**Comment by LinuxOnTheDesktop** [comment]
I tried to pin the flatpak of KSystemLog and I could not - not even having created a launcher.

If the Grouped Window List Applet is to remain in Mint (and it should, for it provided basic functionality) then, if I may say so, it needs a maintainer - and I think it lacks one at the moment.

**Comment by jaszhix2** [comment]
@clefebvre Can you please unban my account so I can unsubscribe from these threads? Linux Mint issues are congesting my Github feed. Thanks.

**Comment by MenthaSuaveolens** [comment]
LM 19.3 - all updates installed

Gimp-issue is partially fixed, but the issue still happens when GIMP is started from Xviewer!

Steps to reproduce
1. Open Nemo and display image with Xviewer 2.4.3
2. In Xviewer : Open image with GIMP => New temp pin is created (see screenshot)
![DoubleGimpEntries](https://user-images.githubusercontent.com/47174957/74916370-03c39200-53c6-11ea-92f1-092dcd5cd08f.png)
